### PR TITLE
Add keywork arg to request #2283

### PR DESCRIPTION
**Pull request by fvlima**
Make all parameters after method and url for urllib3.request() keyword only by adding *

Closes https://github.com/urllib3/urllib3/issues/2283

### CODE REVIEW

File: src/urllib3/request.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def request(
         self,
         method: str,
         url: str,
+        *,
```
Comment (sethmlarson) [inline src/urllib3/request.py:68]: We can't change this one unfortunately, backwards compatibility needs to be maintained for existing APIs.

File: test/test_request.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pytest
```
Comment (sethmlarson) [inline test/test_request.py:1]: The test cases look good but I think there's an existing place for tests of `urllib3.request()`

File: test/test_request.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+from urllib3 import request
+
+from dummyserver.testcase import HTTPDummyServerTestCase
+
+
+class TestRequest(HTTPDummyServerTestCase):
+    @classmethod
```
Comment (sethmlarson) [inline test/test_request.py:9]: I don't think we need the `setup_method`

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this! A few comments for you then we can get this merged.

File: test/test_request.py (hunk 2 above)
Comment (fvlima) [reply test/test_request.py:1]: Great, can you show me? I have doubts to the best place to put this test

File: src/urllib3/request.py (hunk 1 above)
Comment (fvlima) [reply src/urllib3/request.py:68]: Ok, I forgot about the backward compatibility... removing it

File: test/test_request.py (hunk 2 above)
Comment (sethmlarson) [reply test/test_request.py:1]: In the file `test/with_dummyserver/test_poolmanager.py` search for `test_top_level_request`. Under that test case should work!

Comment (sethmlarson) [reply test/test_request.py:1]: We only need the extra `TypeError` case btw, the positive case is already there.

File: test/test_request.py (hunk 3 above)
Comment (fvlima) [reply test/test_request.py]: Done

File: test/test_request.py (hunk 2 above)
Comment (fvlima) [reply test/test_request.py]: Done

File: src/urllib3/request.py (hunk 1 above)
Comment (fvlima) [reply src/urllib3/request.py]: Done

**Review by sethmlarson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**